src/clover/__utils__.py

```diff
@@ -1906,6 +1906,55 @@ def from_dict(cls, hot_water_inputs: Dict[str, Any], logger: logging.Logger) ->
         )
 
 
+class GridType(enum.Enum):
+    """
+    Specifies the type of grid being considered.
+
+    - Current draw:
+        Denotes that the grid is based on the current draw so Diesel Generator.
+    - Daily power:
+        Denotes that the grid is based on the daily power so EDL.
+
+    """
+    CURRENT_DRAW = "current_draw"
+    DAILY_POWER = "daily_power"
+
+@dataclasses.dataclass
+class GridTier:
+    """
+    Specifies the grid tier following an upper bound threshold.
+
+    - Upper Bound type:
+        Denotes the type of the upper bound tier.
+    - Upper Bound value:
+        Denotes the value of the upper bound tier.
+    - costs:
+        Denotes the costs associated with every tier.
+    
+    """
+
+    upper_bound_type: GridType
```

Comment:
## 👏 This looks really good!
:art: One thing would be to move the `upper_bound_type` up to the `Grid` class I think, let me know what your thoughts are :smile:

Reply:
Let's discuss this today